### PR TITLE
backport: bitcoin#14734, #15709, #15718, #16250, #16267, #16325, #16334, #16352, #16464

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
Follow-up missing backports from v18-v19. Some of them have no dash-relevant code or functionally, but still backported related docs/fixes/refactorings from them. 

## What was done?
In some backports I tried to unify dash's related code with bitcoin's implementation such as bitcoin#16598.
Other backports such as bitcoin#16250 or bitcoin#16325 have not dash-relevant feature/code, but they still have useful changes to backport such as refactorings or extra checks or documentation improvements.

- bitcoin#15718
- bitcoin#16250
- bitcoin#14734
- bitcoin#16325
- bitcoin#16352
- bitcoin#16267
- bitcoin#16334
- bitcoin#16464
- bitcoin#15709

## How Has This Been Tested?
Run unit/functional tests

## Breaking Changes
No breaking changes

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by knst** [comment]
> I have made corresponding changes to the documentation

There are changes in doc:
 - no dbus library in depends anymore
 - added an extra dependency for linux build (zmq)
Should any extra docs such as docs.dash.org be updated also for it?

-------------

`test/lint/lint-python-mutable-default-parameters.sh`
There is a new linter. Should I update also something extra in our CI/CD to make it enabled? `lint-all.sh` should run it by default so far as I know, is it enough?

**Comment by thephez** [comment]
> > I have made corresponding changes to the documentation
> 
> There are changes in doc:
> 
> * no dbus library in depends anymore
> * added an extra dependency for linux build (zmq)
> 
> Should any extra docs such as docs.dash.org be updated also for it?

I can't think of anywhere on docs.dash.org we would need to update on this one. We have a page on building, but it refers back to this repository.

File: src/rpc/rawtransaction_util.cpp [hunk 1]
```diff
@@ -207,6 +207,9 @@ void ParsePrevouts(const UniValue& prevTxsUnival, FillableSigningProvider* keyst
                         throw JSONRPCError(RPC_INVALID_PARAMETER, "redeemScript does not match scriptPubKey");
                     }
                 }
+                if (rs.isNull()) {
+                    throw JSONRPCError(RPC_INVALID_PARAMETER, "Missing redeemScript");
```
Comment (UdjinM6) [inline src/rpc/rawtransaction_util.cpp:211]: 16250: nit: partial, need 13072 for changes in `test/functional/rpc_createmultisig.py`

**Review by UdjinM6 — APPROVED** [review]
utACK with one nit

File: src/rpc/rawtransaction_util.cpp (hunk 1 above)
Comment (knst) [reply src/rpc/rawtransaction_util.cpp:211]: I am preparing new PR with cummulative changes for `test/functional/rpc_createmultisig.py`, will be done in next week probably, including 16250, 16026, 14180, 13541 and others

**Review by kittywhiskers — APPROVED** [review]
utACK

File: src/rpc/rawtransaction_util.cpp (hunk 1 above)
Comment (knst) [reply src/rpc/rawtransaction_util.cpp:211]: There is missing changes: https://github.com/dashpay/dash/pull/5278

**Pull request merged**